src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.ts

```diff
@@ -102,9 +102,14 @@ export class SamHierarchicalTreeGridComponent implements OnInit {
       this.columnHeaderText.push(item.headerText);
     });
     this.displayedColumns = [...this.displayedColumns, ...this.columnFieldName];
+
+  }
+
+  ngAfterViewChecked(): void {
```

Comment:
@thomastighe This can be pretty performance intensive. Can you look at changing the ChangeDetectionStrategy to OnPush and seeing if any issues come up? That would only trigger the lifecycle hooks when a new value is pushed into the inputs (there are a few other things that would trigger the rerender, I can't remember off the top of my head though).